numba_dpex/dpnp_iface/arrayobj.py

```diff
@@ -158,24 +159,29 @@ def build_dpnp_ndarray(
             filter selector string, an instance of :class:`dpctl.SyclDevice`
             corresponding to a non-partitioned SYCL device, an instance of
             :class:`dpctl.SyclQueue`, or a `Device` object returnedby
-            `dpctl.tensor.usm_array.device`. Default: `"unknwon"`.
-        queue (:class:`dpctl.SyclQueue`, optional): Not supported.
-            Default: `None`.
+            `dpctl.tensor.usm_array.device`. Default: `None`.
+        sycl_queue (:class:`dpctl.SyclQueue`, optional): The SYCL queue to use
+            for output array allocation and copying. sycl_queue and device
+            are exclusive keywords, i.e. use one or another. If both are
+            specified, a TypeError is raised unless both imply the same
+            underlying SYCL queue to be used. If both are None, a cached
+            queue targeting default-selected device is used for allocation
+            and copying. Default: `None`.
 
     Raises:
-        errors.TypingError: If `sycl_queue` is provided for some reason.
+        errors.TypingError: If both `device` and `sycl_queue` are provided.
 
     Returns:
         DpnpNdArray: The Numba type to represent an dpnp.ndarray.
             The type has the same structure as USMNdArray used to
             represent dpctl.tensor.usm_ndarray.
     """
-    if queue and not isinstance(queue, types.misc.Omitted):
+
+    if sycl_queue and not isinstance(sycl_queue, types.misc.Omitted) and device:
```

Comment:
Can be removed as the check is already inside the `UsmNdArray` constructor.

Reply:
> Can be removed as the check is already inside the `UsmNdArray` constructor.

We can actually remove these checks in all `ol_dpnp_<functions>()` too, we are doing this same check on three levels, which we shouldn't do.